tests/components/frontier_silicon/test_config_flow.py

```diff
@@ -0,0 +1,312 @@
+"""Test the Frontier Silicon config flow."""
+from unittest.mock import patch
+
+from afsapi import ConnectionError, InvalidPinException
+
+from homeassistant import config_entries
+from homeassistant.components.frontier_silicon.const import DOMAIN
+from homeassistant.const import CONF_HOST, CONF_NAME, CONF_PIN, CONF_PORT
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+
+VALID_DEVICE_URL_PATCH = patch(
+    "afsapi.AFSAPI.get_webfsapi_endpoint",
+    return_value="http://1.1.1.1:80/webfsapi",
+)
```

Comment:
These constant make it very hard to read.
If you want to use constants, I think it would be clearer to make just the module name a constant, and use `path(CONSTANT, return_value=...)` inside the tests.